spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -322,6 +323,12 @@ public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
 		this.beanFactory = (ConfigurableListableBeanFactory) beanFactory;
 	}
 
+	private void configureMessageChannels(MessageChannel messageChannel, String channelName) {
+		for (MessageChannelConfigurer messageChannelConfigurer: messageChannelConfigurers) {
```

Comment:
technically, this should have a null-check for the list, but a bigger issue is that we really should not be relying on field-injection in this class at all